flume-ng-channels/flume-kafka-channel/src/test/java/org/apache/flume/channel/kafka/ChannelCallbackTest.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flume.channel.kafka;
+
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ChannelCallbackTest {
+
+  //note that this is log4j backing the slf4j logger
+  //slf4j does not give an API to change log level at runtime
+  private static final Logger CHANNEL_BACKING_LOGGER = Logger.getLogger(ChannelCallback.class);
+
+  Level originalLevel;
+
+  @Before
+  public void setup() {
+    originalLevel = CHANNEL_BACKING_LOGGER.getLevel();
+  }
+
+  @After
+  public void cleanup() {
+    CHANNEL_BACKING_LOGGER.setLevel(originalLevel);
+  }
+
+  private final ChannelCallback channelCallback = new ChannelCallback(0, System.currentTimeMillis());
+  private final RecordMetadata metadata
+          = new RecordMetadata(new TopicPartition("TEST-TOPIC", 0), 0, 0);
+  private final Exception exception = new Exception("TEST");
+
+  @Test
+  public void onCompletion() throws Exception {
```

Comment:
Could you please add some comment here, maybe with a reference to the Jira issue? Knowing the background helps to understand the test and that the expected behaviour is not to throw an exception but without that knowledge it's quite baffling.
It would be even better to wrap the `onCompletion` calls to a `try-catch` block and in case of an (unwanted) NPEx fail the test with an `Assert.fail()`.

Please also rename the test method name to `testOnCompletion` as most of the test methods in Flume have the `test` prefix. (Or to some more self-explaining like `testOnCompletionLoggingNoException` - could be better but this is the best I could come up with)